Client/Frontend/Settings/AppSettingsTableViewController.swift

```diff
@@ -143,4 +148,31 @@ class AppSettingsTableViewController: SettingsTableViewController {
 
         return super.tableView(tableView, heightForRowAtIndexPath: indexPath)
     }
+
+    override func tableView(tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
+        if !profile.hasAccount() {
+            let headerView = tableView.dequeueReusableHeaderFooterViewWithIdentifier(SectionHeaderIdentifier) as! SettingsTableSectionHeaderFooterView
+            let sectionSetting = settings[section]
+            headerView.titleLabel.text = sectionSetting.title?.string
+
+            // Hide the bottom border for the Sign In to Firefox value prop
+            if section == 1 {
+                headerView.titleAlignment = .Top
+                headerView.titleLabel.numberOfLines = 0
+                headerView.showBottomBorder = false
+                headerView.titleLabel.snp_updateConstraints { make in
+                    make.right.equalTo(headerView).offset(-50)
+                }
+                return headerView
+            }
+
+            // Hide the top border for the General section header when the user is not signed in.
+            if section == 2 {
+                headerView.showTopBorder = false
+                return headerView
+            }
```

Comment:
Nit: Having the two `return headerView` is redundant. Can probably clean this up to be:

```swift
switch section {
case 1:
...
case 2:
...
}
return headerView
```